packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -37,10 +37,44 @@ import type {
 } from "./detachedFieldIndexTypes.js";
 import { makeDetachedFieldIndexCodec } from "./detachedFieldIndexCodecs.js";
 
+/**
+ * Readonly interface for {@link DetachedFieldIndex}.
+ */
+export interface ReadOnlyDetachedFieldIndex {
+	/**
+	 * Creates a deep clone of this DetachedFieldIndex.
+	 */
+	clone(): DetachedFieldIndex;
+
+	/**
+	 * Returns a field key for the given ID.
+	 * This does not save the field key on the index. To do so, call {@link createEntry}.
+	 */
+	toFieldKey(id: ForestRootId): FieldKey;
+
+	/**
+	 * Returns the FieldKey associated with the given id.
```

Comment:
Nit: In the method above, we use "field key" rather than "FieldKey". Probably good to be consistent. And if we use the type-name, I always recommend inline code syntax (\`FieldKey\`) or a link.